src/components/common/GProfilerModal.tsx

```diff
@@ -0,0 +1,462 @@
+import { withRouter, RouteComponentProps } from 'react-router-dom';
+
+import React, { Component } from 'react';
+import {
+    Button,
+    Modal,
+    Form,
+    Checkbox,
+    Table,
+    Input,
+    Select,
+    Label,
+    Dropdown,
+} from 'semantic-ui-react';
+
+interface IGProfilerPopupState {
+    error: string;
+    showModal: boolean;
+    topNumFeatures: number[];
+    availableOrganisms: object[];
+    selectedOrganism: string;
+    selectedSortBy: string;
+    gProfilerToken: string;
+}
+
+interface IGProfilerPopupProps {
+    featureMetadata: any;
+    availableSortBy: object[];
+}
+
+const INITIAL_STATE = {
+    error: null,
+    showModal: false,
+    topNumFeatures: [],
+    availableOrganisms: [],
+    selectedOrganism: null,
+    selectedSortBy: null,
+    gProfilerToken: null,
+};
+
+class GProfilerPopup extends Component<
+    IGProfilerPopupProps & RouteComponentProps,
+    IGProfilerPopupState
+> {
+    featureMetadata: any;
+    featureMetricTable: any;
+    availableSortBy: object[];
```

Comment:
Can these be made `private readonly` and the types be made more specific please? They may not even be necessary, e.g. `availableSortBy` could be derived in a reducer.

Reply:
~~reducer~~ selector.